lib/beaker-hostgenerator/generator.rb

```diff
@@ -77,34 +77,21 @@ def generate(tokens, options)
       return config.to_yaml
     end
 
-    def __get_role_config(role)
-      begin
-        r = BeakerHostGenerator::Roles.new
-        m = r.method(role)
-      rescue NameError
-        return {}
-      end
-
-      return m.call
-    end
-
-    def __parse_node_info_token(token)
-      node_info = NODE_REGEX.match(token)
-
-      if node_info
-        node_info = Hash[ node_info.names.zip( node_info.captures ) ]
+    def __generate_host_roles!(host_config, node_info, options)
+      if not options[:disable_default_role]
+        host_config['roles'] = ['agent']
```

Comment:
Similar questions here about action-at-a-distance when modifying passed-in hashes. Maybe copy and returned the result instead?